src/implementation/Client/HTTPClient/HTTPClient.ts

```diff
@@ -144,50 +145,40 @@ export default class HTTPClient implements IClient {
     params?: THTTPExecuteParams | undefined | null,
     requiresInitialization = true,
   ): Promise<object | string> {
-    if (!params || typeof params !== "object") {
-      params = {
-        method: "GET",
-      };
-    }
+    const clientOptions: RequestInit = {};
 
-    if (!params?.headers) {
-      params.headers = {};
-    }
+    // Set Method
+    clientOptions.method = params?.method.toLocaleUpperCase() || (params?.body ? "POST" : "GET");
+
+    // Set Headers
+    clientOptions.headers = params?.headers ?? {};
 
     if (this.options.daprApiToken) {
-      params.headers["dapr-api-token"] = this.options.daprApiToken;
+      clientOptions.headers["dapr-api-token"] = this.options.daprApiToken;
     }
 
-    params.headers["user-agent"] = `dapr-sdk-js/v${SDK_VERSION} http/1`;
+    clientOptions.headers["user-agent"] = `dapr-sdk-js/v${SDK_VERSION} http/1`;
 
-    if (!params?.method) {
-      params.method = "GET";
-    }
+    // Set Body and Content-Type Header
+    if (params?.body) {
+      const { serializedData, contentType } = SerializerUtil.serializeHttp(params?.body);
 
-    if (params?.body && !params?.headers["Content-Type"]) {
-      switch (typeof params?.body) {
-        case "object":
-          params.headers["Content-Type"] = "application/json";
-          params.body = JSON.stringify(params?.body);
-          break;
-        case "string":
-          params.headers["Content-Type"] = "text/plain";
-          break;
-        default:
-          this.logger.warn(`Unknown body type: ${typeof params?.body}, defaulting to "text/plain"`);
-          params.headers["Content-Type"] = "text/plain";
-          break;
+      // Don't overwrite it
+      if (!params?.headers?.["Content-Type"]) {
+        clientOptions.headers["Content-Type"] = contentType;
       }
+
+      clientOptions.body = serializedData;
     }
 
     const urlFull = url.startsWith("http") ? url : `${this.clientUrl}${url}`;
     const agent = urlFull.startsWith("https") ? HTTPClient.httpsAgent : HTTPClient.httpAgent;
-    params.agent = agent;
+    clientOptions.agent = agent;
 
-    this.logger.debug(`Fetching ${params.method} ${urlFull} with body: (${params.body})`);
+    this.logger.debug(`Fetching ${clientOptions.method} ${urlFull} with body: (${clientOptions.body})`);
```

Comment:
All client options in debug logs?

Reply:
This should be clientOptions already in the latest version?

Reply:
I meant, `this.logger.debug("Fetching with clientOptions: ${JSON.stringify(clientOptions)}");` or something.

Reply:
Ah, it just prints the method and body and it's quite useful as it then shows what is going wrong in the execute calls